### ISSUE TITLE
Alignment auto-crop messes with I0 ROI

### ISSUE THREAD

**Issue by benajamin** [mlerotic/spectromicroscopy#24]
When I set an I0 (from histogram) and then align the stack (with "FFT cross-correlation") while using the "Auto-crop stack" option (which is cropping by 3 pixels), then I get the following error:

```
Traceback (most recent call last):
  File "/home/watts/dev/mantis/devel_branch/spectromicroscopy/mantis_xray/mantis_qt.py", line 13294, in OnAccept
    self.parent.page1.specfig.ClearandReload()
  File "/home/watts/dev/mantis/devel_branch/spectromicroscopy/mantis_xray/mantis_qt.py", line 15682, in ClearandReload
    self.loadNewSpectrum()
  File "/home/watts/dev/mantis/devel_branch/spectromicroscopy/mantis_xray/mantis_qt.py", line 15958, in loadNewSpectrum
    self.parent.absimgfig.OnROIShapeChanged(self.parent.ROIShapeBox.currentText())
  File "/home/watts/dev/mantis/devel_branch/spectromicroscopy/mantis_xray/mantis_qt.py", line 16343, in OnROIShapeChanged
    self.parent.specfig.updatePlotDataOnROIShapeChange(shape)
  File "/home/watts/dev/mantis/devel_branch/spectromicroscopy/mantis_xray/mantis_qt.py", line 16013, in updatePlotDataOnROIShapeChange
    self.drawROImask(mask, color=color)
  File "/home/watts/dev/mantis/devel_branch/spectromicroscopy/mantis_xray/mantis_qt.py", line 16169, in drawROImask
    roirgba[indices] = color
IndexError: boolean index did not match indexed array along dimension 0; dimension is 100 but corresponding boolean dimension is 97
```

I suspect that the ROI of the I0 is extending into the region being cropped and this is not being handled properly.

If I use the "Crop stack 3D" routine and crop the I0 area (either partially, or completely), then I don't get any error.

**Comment by Grumium** [comment]
A good one! Funny thing is, I already had a "#ToDo: How to handle aligned i0 data?" action item on this for a long time. 
Fixed in c01d6929ee81e61269c7c279d583f555c66a3245

**Comment by Grumium** [comment]
And a second one: 
793b88b2c30b6d98c94ccb81376452dc8ccebcd5

Branches merged and in sync: 38e1311c83c6f4e4e9d81c8a5e1342538fc0018e
